src/components/Observations/Toolbar.js

```diff
@@ -1,53 +1,139 @@
 // @flow
 
-import { Pressable, View } from "components/styledComponents";
+import { Text, TouchableOpacity, View } from "components/styledComponents";
+import { t } from "i18next";
 import { ObsEditContext } from "providers/contexts";
 import type { Node } from "react";
 import React, { useContext } from "react";
-import { ActivityIndicator } from "react-native";
+import { ProgressBar } from "react-native-paper";
 import IconMaterial from "react-native-vector-icons/MaterialIcons";
 import useCurrentUser from "sharedHooks/useCurrentUser";
+import useLocalObservations from "sharedHooks/useLocalObservations";
+import useNumUnuploadedObservations from "sharedHooks/useNumUnuploadedObservations";
+import useUploadObservations from "sharedHooks/useUploadObservations";
+import colors from "styles/tailwindColors";
 
 type Props = {
-  setView: Function
-  }
+  setView: Function,
+  view: string,
+};
 
-const Toolbar = ( { setView }: Props ): Node => {
+const Toolbar = ( { setView, view }: Props ): Node => {
   const currentUser = useCurrentUser( );
   const obsEditContext = useContext( ObsEditContext );
+  const { allObsToUpload } = useLocalObservations( );
+  const numUnuploadedObs = useNumUnuploadedObservations( );
+  const {
+    stopUpload,
+    uploadInProgress,
+    startUpload,
+    error: uploadError
+  } = useUploadObservations( allObsToUpload );
+
   const loading = obsEditContext?.loading;
   const syncObservations = obsEditContext?.syncObservations;
 
+  const totalObsToUpload = Math.max( allObsToUpload.length, numUnuploadedObs );
+  const progress = ( totalObsToUpload - numUnuploadedObs + 1 ) / ( totalObsToUpload + 1 );
+
+  const getSyncClick = ( ) => {
+    if ( numUnuploadedObs > 0 ) {
+      return startUpload;
+    }
+    return syncObservations;
+  };
+
+  const cleanProgress = ( progressToClean: number ) => {
+    if ( Number.isNaN( progressToClean ) || progressToClean < 0 ) {
+      return 0;
+    }
+
+    return progressToClean;
+  };
+
+  const getStatusText = ( ) => {
+    if ( !uploadInProgress && totalObsToUpload > 0 ) {
+      return t( "UPLOAD-X-OBSERVATIONS", { count: numUnuploadedObs } );
+    }
+    if ( totalObsToUpload > 0 ) {
+      return t( "Uploading-X-Observations", { count: numUnuploadedObs } );
```

Comment:
There are a few other text states mentioned in the issue, where there should be different text for smaller screen sizes. I believe the breakpoint we're looking at for the smallest screen size is 240px of width. Can we add those states in?